### PR TITLE
2.12 release notes

### PR DESCRIPTION
**Pull request by Gallaecio**
There are no short-term plans for a release, this is just to get an idea of what we have in the main branch so far.

### CODE REVIEW

File: docs/news.rst [hunk 1]
```diff
@@ -3,23 +3,475 @@
...
+    :meth:`~scrapy.crawler.Crawler.get_addon`,
+    :meth:`~scrapy.crawler.Crawler.get_downloader_middleware`,
+    :meth:`~scrapy.crawler.Crawler.get_extension`,
+    :meth:`~scrapy.crawler.Crawler.get_item_pipeline`,
+    :meth:`~scrapy.crawler.Crawler.get_spider_middleware`.
+    (:issue:`6181`)
+
+-   :ref:`Automatic throttling <topics-autothrottle>` can now be disabled for a
+    specific slot via its ``throttle`` attribute.
+    (:issue:`6246`)
```
Comment (Gallaecio) [inline docs/news.rst:264]: We need to update this now, and change the wording (the previous wording is my fault).

File: docs/news.rst [hunk 2]
```diff
@@ -3,23 +3,475 @@
...
+
+    .. _brotlicffi: https://github.com/python-hyper/brotlicffi
+
+-   Added :class:`~scrapy.contracts.default.MetadataContract` that sets the
+    request meta.
+    (:issue:`6468`, :issue:`6469`)
+
+-   :func:`scrapy.utils.httpobj.urlparse_cached` is now used in more places
+    instead of :func:`urllib.parse.urlparse`.
+    (:issue:`6228`, :issue:`6229`)
```
Comment (Gallaecio) [inline docs/news.rst:290]: Maybe this fits better in the Improvements section below.

**Review by Gallaecio — COMMENTED** [review]
:heavy_check_mark:

File: docs/news.rst [hunk 3]
```diff
@@ -3,23 +3,476 @@
...
+.. _release-2.12.0:
 
-Scrapy VERSION (YYYY-MM-DD)
----------------------------
+Scrapy 2.12.0 (unreleased)
+--------------------------
 
-New features
-~~~~~~~~~~~~
+Highlights:
```
[the remark this answers is not in the document]
Comment (Gallaecio) [reply docs/news.rst:11]: I think we can drop the stats, getters and type hints. But I am also fine with the current content.

File: docs/news.rst [hunk 4]
```diff
@@ -3,23 +3,476 @@
...
+    :meth:`~scrapy.crawler.Crawler.get_downloader_middleware`,
+    :meth:`~scrapy.crawler.Crawler.get_extension`,
+    :meth:`~scrapy.crawler.Crawler.get_item_pipeline`,
+    :meth:`~scrapy.crawler.Crawler.get_spider_middleware`.
+    (:issue:`6181`)
+
+-   :ref:`Automatic throttling <topics-autothrottle>` can now be disabled for a
+    specific request via its :reqmeta:`autothrottle_dont_adjust_delay` meta
+    key.
+    (:issue:`6246`, :issue:`6527`)
```
Comment (Gallaecio) [inline docs/news.rst:265]: Let me give this a go…

Comment (Gallaecio) [reply docs/news.rst:265]:
```suggestion
-   Slot delay updates by the :ref:`AutoThrottle extension 
    <topics-autothrottle>` based on response latencies can now be disabled for 
    specific requests via the :reqmeta:`autothrottle_dont_adjust_delay` meta
    key.
    (:issue:`6246`, :issue:`6527`)
```

**Comment by Gallaecio** [comment]
I don’t think https://github.com/scrapy/scrapy/pull/6226#discussion_r1832946651 is necessary, but I think you might have just forgotten about it.

File: scrapy/pipelines/media.py [hunk 5]
```diff
@@ -149,6 +149,15 @@ def from_crawler(cls, crawler: Crawler) -> Self:
         pipe: Self
         if hasattr(cls, "from_settings"):
             pipe = cls.from_settings(crawler.settings)  # type: ignore[attr-defined]
+            warnings.warn(
+                f"{cls.__qualname__} has from_settings() and either doesn't have"
```
Comment (Gallaecio) [inline scrapy/pipelines/media.py:153]: `global_object_name`?

**Comment by Laerte** [comment]
@wRAR @Gallaecio 

For some reason `latest` (the link that you got redirect when accessing: https://scrapy.org/doc/): https://docs.scrapy.org/en/latest/ still pointing to 2.11 (Last updated on Oct 08, 2024.).

It takes some time to update or maybe it failed to build?

**Comment by wRAR** [comment]
@Laerte https://github.com/readthedocs/readthedocs.org/issues/11772

**Comment by wRAR** [comment]
@Laerte this is now fixed

**Comment by Laerte** [comment]
> @Laerte this is now fixed

Thank you!

**Pull request merged**